lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -10,29 +10,40 @@ def initialize(serializer)
 
           def href(value)
             self._href = value
+            nil
           end
 
           def meta(value)
             self._meta = value
+            nil
           end
 
-          def string(value)
-            self._string = value
+          def value(value)
+            if value.respond_to?(:call)
+              string instance_eval(&value)
+              _string || to_hash
```

Comment:
I do not understand those 2 lines though. This will always return `_string`, unless it is falsy, which happens only if – wait, I got it, that's why `meta` and `href` return `nil`.
How about the following:
```ruby
def initialize(serializer, value)
  @object = serializer.object
  @scope = serializer.scope
  if value.respond_to?(:call)
    instance_eval(&value)
  else
    self._string = value
  end
end

def value
  _string || to_hash
end
```